src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityActions.php

```diff
@@ -83,12 +83,14 @@ public function getEnvironment()
         return $this->environment;
     }
 
-    /**
-     * @return bool
-     */
-    public function allowEditAction()
+    public function allowEditAction(): bool
     {
-        if ($this->isPublishedToProduction() || $this->readOnly) {
+        $notEditable =
+            $this->isPublishedToProduction()
+            || $this->readOnly
+            || $this->status === Constants::STATE_REMOVAL_REQUESTED;
```

Comment:
Instead of catching this in a notEditable variable, you could directly return the outcome of this expression.

Reply:
I actually did that first, but found it far less legible to the point that i wanted to add a comment to it to clarify.  So went with this as a middle ground.

Reply:
I can live with that! And it improves debugabillity as well.